modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/MvPartitionStorages.java

```diff
@@ -442,6 +403,8 @@ private void throwExceptionDependingOnOperation(StorageOperation operation, int
             throw new StorageRebalanceException(createStorageInProgressOfFinishRebalanceErrorMessage(partitionId));
         } else if (operation instanceof CleanupStorageOperation) {
             throw new StorageException(createStorageInProgressOfCleanupErrorMessage(partitionId));
+        } else if (operation instanceof CloseStorageOperation || operation.isFinalOperation()) {
```

Comment:
I am glad that there are individual classes for different operations.
But, I am not glad that you have a bunch of `if-else` statements instead of using polymorphism.
What's the point of having classes then?